### PR TITLE
Trigger GH jobs when there are changes in packages/sdk

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Trigger GH jobs when there are changes on `packages/sdk`. The rationale is because other packages depends on the packages from this folder. This is a simplification from our circleci's implementation that has a more complex approach using `scripts/check_if_test_should_run_v2.ts`. I preferred not to replicate same logic because that would require creating the runners for all the jobs even when they are going to be skipped, which would increase time and costs of execution; and also GitHub Actions offer an easier and simpler way to track files changed in the PR/commit inside the workflow logic.
With this approach, there may be some tests executed when they are not strictly required, but I think should not skip a test when one of its "local" dependencies has changed.

Also added the condition to run all the test for the commits to specific branches (`master`, and any branch containing `staging` or `production`)

### CODE REVIEW

File: .github/workflows/circleci.yml [hunk 1]
```diff
@@ -389,7 +396,8 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies, contractkit-tests]
     if: |
-      contains(needs.install-dependencies.outputs.all_modified_files, 'packages/identity') ||
+      github.base_ref == 'master' || contains(github.base_ref, 'staging') || contains(github.base_ref, 'production') ||
+      contains(needs.install-dependencies.outputs.all_modified_files, 'packages/sdk') ||
```
Comment (soloseng) [inline .github/workflows/circleci.yml:409]: This should also run when changes are made to:
- `@celo/phone-number-privacy-common`
- `@celo/dev-utils`
- `@celo/protocol`
- `@celo/typescript`

**Comment by aaronmgdr** [comment]
brilliant! and all the checks pass 🌙

**Review by aaronmgdr — APPROVED** [review]

File: .github/workflows/circleci.yml [hunk 2]
```diff
@@ -361,7 +367,8 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies]
     if: |
```
Comment (soloseng) [inline .github/workflows/circleci.yml:372]: This should also run when changes are made to:
- `@celo/protocol`
- `@celo/dev-utils`
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 3]
```diff
@@ -417,7 +425,8 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies, contractkit-tests]
     if: |
-      contains(needs.install-dependencies.outputs.all_modified_files, 'packages/sdk/transactions-uri') ||
+      github.base_ref == 'master' || contains(github.base_ref, 'staging') || contains(github.base_ref, 'production') ||
+      contains(needs.install-dependencies.outputs.all_modified_files, 'packages/sdk') ||
```
Comment (soloseng) [inline .github/workflows/circleci.yml:441]: This should also run when changes are made to:
- `@celo/protocol`
- `@celo/typescript`
- `@celo/dev-utils`

File: .github/workflows/circleci.yml [hunk 4]
```diff
@@ -450,7 +459,9 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies]
     if: |
+      github.base_ref == 'master' || contains(github.base_ref, 'staging') || contains(github.base_ref, 'production') ||
       contains(needs.install-dependencies.outputs.all_modified_files, 'packages/cli') ||
+      contains(needs.install-dependencies.outputs.all_modified_files, 'packages/sdk') ||
```
Comment (soloseng) [inline .github/workflows/circleci.yml:480]: This should also run when changes are made to:
- `@celo/phone-number-privacy-common`
- `@celo/dev-utils`
- `@celo/protocol`
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 5]
```diff
@@ -517,6 +529,7 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies]
     if: |
+      github.base_ref == 'master' || contains(github.base_ref, 'staging') || contains(github.base_ref, 'production') ||
       contains(needs.install-dependencies.outputs.all_modified_files, 'packages/sdk') ||
```
Comment (soloseng) [inline .github/workflows/circleci.yml:550]: This should also run when changes are made to:
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 6]
```diff
@@ -551,6 +564,7 @@ jobs:
...
```
Comment (soloseng) [inline .github/workflows/circleci.yml:586]: This should also run when changes are made to:
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 7]
```diff
@@ -674,7 +689,9 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies, lint-checks]
     if: |
+      github.base_ref == 'master' || contains(github.base_ref, 'staging') || contains(github.base_ref, 'production') ||
       contains(needs.install-dependencies.outputs.all_modified_files, 'packages/phone-number-privacy') ||
```
Comment (soloseng) [inline .github/workflows/circleci.yml:715]: This should also run when changes are made to:
- @celo/protocol

File: .github/workflows/circleci.yml [hunk 8]
```diff
@@ -584,9 +598,10 @@ jobs:
     timeout-minutes: 60
     needs: [install-dependencies, lint-checks, contractkit-tests]
     if: |
+      github.base_ref == 'master' || contains(github.base_ref, 'staging') || contains(github.base_ref, 'production') ||
       contains(needs.install-dependencies.outputs.all_modified_files, 'packages/protocol') ||
```
Comment (soloseng) [inline .github/workflows/circleci.yml]: This should also run when changes are made to:
- `@celo/phone-number-privacy-common`
- `@celo/dev-utils`
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 9]
```diff
@@ -215,7 +215,9 @@ jobs:
     # needs: [install-dependencies, lint-checks]
     needs: [install-dependencies]
     if: |
```
Comment (soloseng) [inline .github/workflows/circleci.yml:217]: This should also run when changes are made to:
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 10]
```diff
@@ -256,7 +258,9 @@ jobs:
     timeout-minutes: 30
     needs: [install-dependencies, lint-checks, pre-protocol-test-release]
     if: |
```
Comment (soloseng) [inline .github/workflows/circleci.yml:261]: This should also run when changes are made to:
- `@celo/typescript`

File: .github/workflows/circleci.yml [hunk 11]
```diff
@@ -287,7 +291,9 @@ jobs:
     timeout-minutes: 60
     needs: [install-dependencies, lint-checks]
     if: |
```
Comment (soloseng) [inline .github/workflows/circleci.yml:295]: This should also run when changes are made to:
- `@celo/typescript`

**Review by soloseng — COMMENTED** [review]
I think this should work. I left a few comment that would ensure packages are tested when nested dependencies are changed.

**Comment by soloseng** [comment]
I just noticed that the test only runs based only on the last pushed commit changes.

for example, If I made changes to `@celo/protocol`, committed without pushing then made a change to `circleci.yml`, commit and push all my changes. `protocol` tests are skipped.

The expected behavior should be that tests are run for all the previous changes since last pushed commit.

This could be out of the scope of this PR, but I figured i'd mention it here.

**Comment by jcortejoso** [comment]
> I just noticed that the test only runs based only on the last pushed commit changes.
> 
> for example, If I made changes to `@celo/protocol`, committed without pushing then made a change to `circleci.yml`, commit and push all my changes. `protocol` tests are skipped.
> 
> The expected behavior should be that tests are run for all the previous changes since last pushed commit.
> 
> This could be out of the scope of this PR, but I figured i'd mention it here.

It should be as you describe (at least based on action readme), but also noticed yesterday. I'll check it.
image

**Review by soloseng — APPROVED** [review]
🚀 LGTM

**Pull request merged**